### PR TITLE
feat: update migrate script to remap gencode terms

### PR DESCRIPTION
**Pull request by joyceyan**
## Reason for Change

https://app.zenhub.com/workspaces/single-cell-5e2a191dad828d52cc78b028/issues/gh/chanzuckerberg/single-cell-curation/726

## Changes

- adds the remapped gencode terms

## Testing

added tests to `test_migrate.py` and `test_utils.`py

### CODE REVIEW

File: cellxgene_schema_cli/cellxgene_schema/migrate.py [hunk 1]
```diff
@@ -349,6 +365,9 @@ def migrate(input_file, output_file, collection_id, dataset_id):
 
     # AUTOMATED, DO NOT CHANGE -- IF GENCODE UPDATED, DEPRECATED FEATURE FILTERING ALGORITHM WILL GO HERE.
     if DEPRECATED_FEATURE_IDS:
-        dataset = utils.remove_deprecated_features(dataset, DEPRECATED_FEATURE_IDS)
+        dataset = utils.remove_deprecated_features(adata=dataset, deprecated=DEPRECATED_FEATURE_IDS)
+
+    # Manually remap v38 terms to v44
+    dataset = utils.remap_deprecated_features(adata=dataset, remapped_terms=GENCODE_MAPPER)
 
-    dataset.write(output_file, compression="gzip")
```
Comment (joyceyan) [inline cellxgene_schema_cli/cellxgene_schema/migrate.py:354]: i'm not entirely sure what this change is - i didn't intentionally alter this line at least lol

File: cellxgene_schema_cli/cellxgene_schema/migrate.py [hunk 2]
```diff
@@ -316,6 +316,22 @@
...
+    'ENSG00000215271': 'ENSG00000290292',
+    'ENSG00000225932': 'ENSG00000288784',
+    'ENSG00000244693': 'ENSG00000289604',
+    'ENSG00000203812': 'ENSG00000288825',
+    'ENSG00000272196': 'ENSG00000288859',
+    'ENSG00000263464': 'ENSG00000288867',
+    'ENSG00000256374': 'ENSG00000289549',
+    'ENSG00000288546': 'ENSG00000289084',
+    'ENSG00000205485': 'ENSG00000290873'
+}
```
Comment (danieljhegeman) [inline cellxgene_schema_cli/cellxgene_schema/migrate.py:334]: 👀 ✅

File: cellxgene_schema_cli/cellxgene_schema/migrate.py (hunk 1 above)
Comment (danieljhegeman) [reply cellxgene_schema_cli/cellxgene_schema/migrate.py:354]: Every once in a while github makes mistakes in line-grouping and detecting diffs. I've seen it render an enormous diff for a file when in actuality only one line had changed.

File: cellxgene_schema_cli/cellxgene_schema/utils.py [hunk 3]
```diff
@@ -40,7 +40,7 @@ def map_ontology_term(dataframe, ontology_name, map_from_column, update_map):
     dataframe[column_name] = dataframe[column_name].cat.remove_unused_categories()
 
 
-def remove_deprecated_features(adata: ad.AnnData, deprecated: List[str]) -> ad.AnnData:
+def remove_deprecated_features(*, adata: ad.AnnData, deprecated: List[str]) -> ad.AnnData:
```
Comment (danieljhegeman) [inline cellxgene_schema_cli/cellxgene_schema/utils.py:43]: Huh. The `*` marking the boundary between positional vs kwargs is new to me. Cool 🤙

File: cellxgene_schema_cli/cellxgene_schema/utils.py [hunk 4]
```diff
@@ -54,6 +54,18 @@ def remove_deprecated_features(adata: ad.AnnData, deprecated: List[str]) -> ad.A
     return adata
 
 
+def remap_deprecated_features(*, adata: ad.AnnData, remapped_terms: Dict[str, str]) -> ad.AnnData:
```
Comment (danieljhegeman) [inline cellxgene_schema_cli/cellxgene_schema/utils.py:57]: want to reconcile the function name with the second kwarg var name? 

i.e. `remap_deprecated_features` and `remapped_features`
OR
`remap_deprecated_terms` and `remapped_terms`

Unless there's an abstract reason (literally) for the names precisely as they already are

**Review by danieljhegeman — APPROVED** [review]
Tempted to request that we add `asserts` at the outset of each unit test you added to establish the pre-remapping condition, i.e. "before", but I suppose it's fine as they're predefined shared fixtures that we can rely on.

Or can we? You make the call. I'm fine as-is.

**Comment by joyceyan** [comment]
> Tempted to request that we add `asserts` at the outset of each unit test you added to establish the pre-remapping condition, i.e. "before", but I suppose it's fine as they're predefined shared fixtures that we can rely on.
> 
> Or can we? You make the call. I'm fine as-is.

I think that's a reasonable change just to make the tests easier to read / understand

**Pull request merged**